test/Integration/IntegrationTestBase.cs

```diff
@@ -164,6 +165,56 @@ private void ExecuteAllScriptsInFolder(string folder)
             }
         }
 
+        /// <summary>
+        /// Build the samples-java and test-java projects.
+        /// </summary>
+        private void BuildJavaFunctionApps()
+        {
+            // To build Java function apps while running tests locally, you need to have Maven installed
+            // and the MAVEN_PATH environment variable set to the path to the mvn.cmd file.
+            // The Java projects are built using the Maven task in ADO pipelines.
+            string mavenPath = Environment.GetEnvironmentVariable("MAVEN_PATH");
+            if (!string.IsNullOrEmpty(mavenPath))
+            {
+                // Build samples-java
+                string samplesJavaPath = Path.Combine(GetPathToBin(), "SqlExtensionSamples", "Java");
+                this.BuildJavaFunctionApp(mavenPath, samplesJavaPath);
+
+                // Build test-java
+                string testJavaPath = Path.Combine(GetPathToBin(), "..", "..", "..", "Integration", "test-java");
+                this.BuildJavaFunctionApp(mavenPath, testJavaPath);
+            }
+        }
+
+        /// <summary>
+        /// Run `mvn clean package` to build the Java function app.
+        /// </summary>
+        private void BuildJavaFunctionApp(string mavenPath, string workingDirectory)
+        {
+            var maven = new Process()
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = mavenPath,
```

Comment:
You shouldn't need to specify the full path to maven now that you're using shell execute. (hopefully at least, this is assuming it's being added to the path and whatnot...)

Try it out and see if we can get rid of all this extra path logic and make things a lot simpler.